InterGraph/interaction_graph.py

```diff
@@ -0,0 +1,404 @@
+import os.path
+import pandas as pd
+import os
+import pickle
+import urllib
+import pandas as pd
+import numpy as np
+import networkx as nx
+import openbabel
+
+from tqdm import tqdm
+from rdkit import Chem
+from scipy.spatial.distance import cdist
+from itertools import product
+from openbabel import openbabel
+
+pdb_raw_d = "../data/" + "pdb_raw"
+
+
+def file_path():
+    global pdb_raw_d
+    if not os.path.exists("../data/" + "data"):
+        os.makedirs("../data/" + "data")
+    if not os.path.exists("../data/" + "pdb_raw"):
+        os.makedirs("../data/" + "pdb_raw")
+
+
+def get_pdb_components(pdb_id):
+    global pdb_raw_d
+
+    pdb_r = urllib.request.urlretrieve(
+        "https://files.rcsb.org/download/{}.pdb".format(pdb_id),
+        "{}/{}.pdb".format(pdb_raw_d, pdb_id),
+    )
+
+
+def parsePDB(lig, filename):
+    prot = []
+    ligs = []
+
+    with open(filename, "r") as f:
+        for l in f:
+            if l.split()[0] == "ATOM":
+                prot += [l]
+            if len(l.split()) > 3 and l.split()[3] == lig:
+                ligs += [l]
+
+    return prot, ligs
+
+
+def write_pdb(prot, lig, filename):
+    with open(filename, "w") as f:
+        for l in prot:
+            f.write(l)
+            f.write("TER\n")
+        for l in lig:
+            f.write(l)
+
+
+def write_ligand_pdb(lig, filename):
+
+    with open(filename, "w") as f:
+        for l in lig:
+            l.split(",")
+            f.write(l)
+            return f
+
+
+def load_structures():
+    raw_data = "data"
+    list_of_pdbcodes = [i for i in os.listdir(raw_data)]
+    Atom_Keys = pd.read_csv("../data/csv/PDB_Atom_Keys.csv", sep=",")
+
+    for pdb in tqdm(list_of_pdbcodes):
+        lig_path = os.path.join(raw_data, pdb, f"{pdb[-3:]}.mol2")
+        protein_path = os.path.join(raw_data, pdb, f"{pdb}.pdb")
+
+        # load the ligand and handle invalid mol2 file
+        try:
+            c_mol = Chem.AddHs(
+                Chem.MolFromMol2File(
+                    lig_path, sanitize=False, removeHs=True, cleanupSubstructures=True
+                ),
+                addCoords=True,
+            )
+            print("Mol from Mol2 conversion succeded  ", lig_path)
+        except:
+            print("Mol from Mol2 conversion failed ", lig_path)
+            continue
+
+        # 6 Angstrom
+        mol_graphs_crystal_6A = {}
+        contacts_6A = GetAtomContacts(
+            protein_path, c_mol, Atom_Keys, distance_cutoff=6.0
+        )
+        graph_c_6 = mol_to_graph(c_mol, contacts_6A, Atom_Keys)
+        mol_graphs_crystal_6A[pdb] = graph_c_6
+        return mol_graphs_crystal_6A
+
+
+def GetAtomType(atom):
```

Comment:
I'd recommend to use the PEP 8 style guide recommendation for naming functions:
https://peps.python.org/pep-0008/#function-and-variable-names

Reply:
I've committed and pushed the suggested changes in this pull request branch